### ISSUE TITLE
WebCrypto.getRandomValues vs crypto.randomBytes

### ISSUE THREAD

**Issue by diasdavid** [libp2p/js-libp2p-crypto#46]
#### crypto.randomBytes

https://nodejs.org/api/crypto.html#crypto_crypto_randombytes_size_callback

returns a Buffer with random bytes given the size

#### WebCrypto.getRandomValues(arr)

https://developer.mozilla.org/en/docs/Web/API/RandomSource/getRandomValues

overwrites an array with randomBytes

**Comment by dignifiedquire** [comment]
are you trying to collect all strange behavior of webcrypto here? ;)

**Comment by diasdavid** [comment]
@dignifiedquire exactly, hopefully it will be useful to enumerate issues in the future

**Comment by dignifiedquire** [comment]
As a side note, the webcrypto api is much more sensible when trying to optimize for speed and allocations. You can manage your own array and fill it with random values, whereas the node api is actually much worse, as it forces allocation of a new buffer when generating random values.

**Comment by diasdavid** [comment]
Still a `WAT`

**Comment by daviddahl** [comment]
See: https://bugzilla.mozilla.org/show_bug.cgi?id=440046#c39 

and

https://bugzilla.mozilla.org/show_bug.cgi?id=673432#c5
